src/styles/main.scss

```
@@ -1,3 +1,58 @@
+@import "./utils/variables.scss";
+
 body {
-  margin: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $size-box-gap;
+
+  padding: 10px;
+  max-width: 6 * $size-box-gap + 7 * $size-box;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+
+  @for $days-in-month from 28 through 31 {
+    &--month-length--#{$days-in-month} :nth-child(n + #{$days-in-month + 1}) {
+      display: none;
+    }
+  }
+
+  &__box {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    height: $size-box;
+    width: $size-box;
+
+    background-color: #eee;
+    box-sizing: border-box;
+    border: 1px solid #000;
+
+    cursor: pointer;
+    transition: 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    @each $day, $number in $weekdays {
+      &--start-day--#{$day} :first-child {
+        margin-left: ($number - 1) * ($size-box + $size-box-gap);
```

if your mapped variable starts with `"mon" 0` you could avoid subtracting :)
```suggestion
        margin-left: $number * ($size-box + $size-box-gap);
```